docker/spec/puppetserver_spec.rb

```diff
@@ -7,9 +7,22 @@
   include Helpers
 
   def puppetserver_health_check(container)
-    result = run_command("docker inspect \"#{container}\" --format '{{.State.Health.Status}}'")
-    status = result[:stdout].chomp
-    STDOUT.puts "queried health status of #{container}: #{status}"
+    empty_response_counter = 0
+    status = ''
+
+    # We intermittently get an empty response from the `docker inspect`
+    # but we don't want this to run forever if the container failed, so
+    # assume that 5 empty responses in a row is a failure case
+    while status.empty? && empty_response_counter < 5
+      result = run_command("docker inspect \"#{container}\" --format '{{.State.Health.Status}}'")
+      status = result[:stdout].chomp
+      if status.empty?
+        empty_response_counter += 1
+        sleep 1
+      end
+      STDOUT.puts "queried health status of #{container}: #{status}"
+    end
```

Comment:
FYI we've been using a time-based timeout in our spec tests, like this: https://github.com/puppetlabs/puppetdb/blob/master/docker/puppetdb/spec/puppetdb_spec.rb#L15-L24

Reply:
Ah yeah, we'll probably refactor this a bit with the shared helpers.

More to come!